framework/src/main/java/org/apache/drill/test/framework/TestDriver.java

```diff
@@ -241,189 +265,234 @@ public int runTests() throws Exception {
           }
         }
       }
+
+      LOG.info("\n" + LINE_BREAK);
+      LOG.info("RUNNING TESTS - ITERATION " + i + " (of " + cmdParam.iterations + ")");
+      LOG.info(LINE_BREAK);
       executor.executeAll(tests);
 
       if (cmdParam.trackMemory) {
-    	  queryMemoryUsage();
-      }
-      List<DrillTest> passingTests = Lists.newArrayList();
-      List<DrillTest> dataVerificationFailures = Lists.newArrayList();
-      List<DrillTest> planVerificationFailures = Lists.newArrayList();
-      List<DrillTest> executionFailures = Lists.newArrayList();
-      List<DrillTest> timeoutFailures = Lists.newArrayList();
-      List<DrillTest> canceledTests = Lists.newArrayList();
-      List<DrillTest> randomFailures = Lists.newArrayList();
-      List<DrillTest> failedCases = Lists.newArrayList();
-      
+        queryMemoryUsage();
+      }
+
+      // Isolating Random Failures
+      List<DrillTest> tempTests = Lists.newArrayList();
       for (DrillTest test : tests) {
-        TestStatus testStatus = test.getTestStatus();
-        if(testStatus!=TestStatus.PASS && testStatus!=TestStatus.CANCELED && cmdParam.skipRandom!=true){
-	   List<DrillTest> tempTests = Lists.newArrayList();
-           tempTests.add(test); 
-           executor.executeAll(tempTests);
-           testStatus = tempTests.get(0).getTestStatus();
-           if(testStatus==TestStatus.PASS){
-	     randomFailures.add(test);
-	   }
-	}
-	switch (testStatus) {
-         case PASS:
-           passingTests.add(test);
-           break;
-         case DATA_VERIFICATION_FAILURE:
-           dataVerificationFailures.add(test);
-           break;
-         case PLAN_VERIFICATION_FAILURE:
-           planVerificationFailures.add(test);
-           break;
-         case EXECUTION_FAILURE:
-           executionFailures.add(test);
-           break;
-         case TIMEOUT:
-           timeoutFailures.add(test);
-           break;
-         case CANCELED:
-           canceledTests.add(test);
-           break;
-         default:
-           executionFailures.add(test);
+        if (test.getTestStatus() != TestStatus.PASS && test.getTestStatus() != TestStatus.CANCELED && cmdParam.skipRandom != true) {
```

Comment:
Cleaned-up logic to identify random failures. Also added a message to indicate the same to console. This makes it clear that the first round of tests are complete and a second round is being done for failed tests to isolate random failures.